src/DataStructures/Tensor/Tensor.hpp

```diff
@@ -498,3 +498,7 @@ std::ostream& operator<<(std::ostream& os,
      << "\n     Multiplicity: " << x.multiplicity(i) << " Index: " << i;
   return os;
 }
+
+// These includes are for functions which we wish to be included with tensor but
+// also depend on tensor
+#include "DataStructures/Tensor/Expressions/Magnitude.hpp"
```

Comment:
Don't do this. It'll result in terrible compile times. Blaze does this and it's really bad.